pandas/core/strings.py

```diff
@@ -202,15 +202,43 @@ def str_count(arr, pat, flags=0):
     """
     Count occurrences of pattern in each string of the Series/Index.
 
+    This function is used to count the number of times a particular regex
+    pattern is repeated in each of the string elements of the
+    :class:`~pandas.Series`.
+
     Parameters
     ----------
-    pat : string, valid regular expression
-    flags : int, default 0 (no flags)
-        re module flags, e.g. re.IGNORECASE
+    pat : str
+        Valid regular expression.
+    flags : int, default 0, meaning no flags
+        Flags for re module, e.g. re.IGNORECASE.
 
     Returns
     -------
     counts : Series/Index of integer values
+
+    Notes
+    -----
+    Some characters need to be escaped when passing in pat.
+    eg. '$' has a special meaning in regex and must be escaped when finding
+    specifically this char.
+
+    Examples
+    --------
+    Take a look at
```

Comment:
Maybe move this link to be under the flags parameter?

Reply:
I like that idea, will do